ExampleMod/Content/NPCs/ExampleBoneMerchant.cs

```diff
@@ -0,0 +1,155 @@
+using ExampleMod.Content.Dusts;
+using ExampleMod.Content.Items;
+using System.Linq;
+using Terraria;
+using Terraria.ID;
+using Terraria.Localization;
+using Terraria.ModLoader;
+using Terraria.Utilities;
+using Terraria.GameContent.Bestiary;
+
+namespace ExampleMod.Content.NPCs
+{
+	/// <summary>
+	/// The main focus of this NPC is to show how to make something similar to the vanilla bone merchant;
+	/// which means that the NPC will act like any other town NPC but won't have a happiness button, won't appear on the minimap,
+	/// and will spawn like an enemy NPC. If you want a tradition town NPC instead, <see cref="ExamplePerson"/>.
+	/// </summary>
+	public class ExampleBoneMerchant : ModNPC
+	{
+		public override void SetStaticDefaults() {
+			Main.npcFrameCount[Type] = 25; // The amount of frames the NPC has
+
+			NPCID.Sets.ExtraFramesCount[Type] = 9; // Generally for Town NPCs, but this is how the NPC does extra things such as sitting in a chair and talking to other NPCs.
+			NPCID.Sets.AttackFrameCount[Type] = 4;
+			NPCID.Sets.DangerDetectRange[Type] = 700; // The amount of pixels away from the center of the npc that it tries to attack enemies.
+			NPCID.Sets.AttackType[Type] = 0;
+			NPCID.Sets.AttackTime[Type] = 90; // The amount of time it takes for the NPC's attack animation to be over once it starts.
+			NPCID.Sets.AttackAverageChance[Type] = 30;
+			NPCID.Sets.HatOffsetY[Type] = 4; // For when a party is active, the party hat spawns at a Y offset.
+
+			// Influences how the NPC looks in the Bestiary
+			NPCID.Sets.NPCBestiaryDrawModifiers drawModifiers = new NPCID.Sets.NPCBestiaryDrawModifiers(0) {
+				Velocity = 1f, // Draws the NPC in the bestiary as if its walking +1 tiles in the x direction
+				Direction = 1 // -1 is left and 1 is right. NPCs are drawn facing the left by default but ExamplePerson will be drawn facing the right
+				// Rotation = MathHelper.ToRadians(180) // You can also change the rotation of an NPC. Rotation is measured in radians
+				// If you want to see an example of manually modifying these when the NPC is drawn, see PreDraw
```

Comment:
The comment regarding PreDraw does not apply to this file (copied from ExamplePerson)